### PR TITLE
Lab2b jordan

### PR DESCRIPTION
**Pull request by jordan-heemskerk**
Here's what I've been working on tonight.

I think I have solved the weird error in your screenshots, which I was able to reproduce. The changes for that are in 529c9da and mostly have to do with only allowing a view change once the current delta view has been acked.

### CODE REVIEW

File: src/viewservice/server.go [hunk 1]
```diff
@@ -25,11 +25,13 @@ type ViewServer struct {
...
 	Acked     bool
 
+	RPCs uint
+
 	start bool
 }
 
 func (vs *ViewServer) GetRPCCount() uint {
-	return 0
+	return vs.RPCs
```
Comment (eburdon) [inline src/viewservice/server.go:34]: :+1:

File: src/viewservice/server.go [hunk 2]
```diff
@@ -74,6 +76,10 @@ func (vs *ViewServer) NextView(primary string, backup string) {
 
 		fmt.Printf("Setting next view %d (%s, %s)\n", vs.DeltaView.Viewnum, vs.DeltaView.Primary, vs.DeltaView.Backup)
 
+		if vs.DeltaView.Primary == vs.DeltaView.Backup && vs.DeltaView.Primary != "" {
+			log.Fatal("Server cannot be primary and backup")
```
Comment (eburdon) [inline src/viewservice/server.go:80]: There's no way to recover from this case? Can we make a guarantee that if p = b, it happened when the backup was promoted  or added, and just clear the backup value?

Comment (jordan-heemskerk) [inline src/viewservice/server.go:80]: Just wanted to catch this at the source. If this happens, something really bad has happened, basically.

**Comment by jordan-heemskerk** [comment]
@eburdon think we can merge this?

**Comment by eburdon** [comment]
@jordan-heemskerk yeah sounds good. Awesome work.

**Pull request merged**